references/segmentation/presets.py

```diff
@@ -1,39 +1,40 @@
 import torch
-import transforms as T
+from torchvision.prototype import features, transforms as T
 
 
-class SegmentationPresetTrain:
-    def __init__(self, *, base_size, crop_size, hflip_prob=0.5, mean=(0.485, 0.456, 0.406), std=(0.229, 0.224, 0.225)):
-        min_size = int(0.5 * base_size)
-        max_size = int(2.0 * base_size)
+class WrapIntoFeatures(T.Transform):
+    def forward(self, sample):
+        image, segmentation_mask = sample
+        return image, features.SegmentationMask(segmentation_mask.squeeze(0), dtype=torch.int64)
+
 
-        trans = [T.RandomResize(min_size, max_size)]
+class SegmentationPresetTrain(T.Compose):
+    def __init__(self, *, base_size, crop_size, hflip_prob=0.5, mean=(0.485, 0.456, 0.406), std=(0.229, 0.224, 0.225)):
+        transforms = [
+            T.ToImageTensor(),
+            WrapIntoFeatures(),
```

Comment:
In contrast to the classification and detection references, we need to wrap before all transformation. This is now needed because the new transformation would take the mask as image otherwise.

Reply:
Wrapping isn't the problem. Calling `ToImageTensor` first is I think. This is because if we run this check, we will be hitting the Tensor backend and not the PIL like we did with the previous tests. Everything should be tested obviously, but I wonder whether we can move the tensor conversion at the end?

Reply:
Yes, that should be doable. I'll send a patch.